### PR TITLE
Repairing position keyword TOP n in SELECT

### PR DESCRIPTION
**Pull request by JosefDohnal**
First (after SELECT) must be keywords TOP or DISTINCT, see
https://msdn.microsoft.com/en-us/library/ms189499.aspx,
<query_specification> ::=

### CODE REVIEW

File: src/Database/Drivers/SqlsrvDriver.php [hunk 1]
```diff
@@ -89,7 +89,8 @@ public function formatLike($value, $pos)
 	public function applyLimit(& $sql, $limit, $offset)
 	{
 		if ($limit >= 0) {
-			$sql = preg_replace('#^\s*(SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
+			/** @see https://msdn.microsoft.com/en-us/library/ms189499.aspx */
+            $sql = preg_replace('#^\s*(SELECT\s*DISTINCT|SELECT\s*ALL|SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
```
Comment (dg) [inline src/Database/Drivers/SqlsrvDriver.php]: There should be SELECT\s+ and tabs

File: src/Database/Drivers/SqlsrvDriver.php [hunk 2]
```diff
@@ -89,7 +89,8 @@ public function formatLike($value, $pos)
 	public function applyLimit(& $sql, $limit, $offset)
 	{
 		if ($limit >= 0) {
-			$sql = preg_replace('#^\s*(SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
+			/** @see https://msdn.microsoft.com/en-us/library/ms189499.aspx */
+            $sql = preg_replace('#^\s*(SELECT\s+DISTINCT|SELECT\s+ALL|SELECT|UPDATE|DELETE)#i', '$0 TOP ' . (int) $limit, $sql, 1, $count);
```
Comment (dg) [inline src/Database/Drivers/SqlsrvDriver.php]: use tabs, not spaces

**Comment by dg** [comment]
Can you squash it to single commit and use commit message like `SqlsrvDriver::applyLimit(): supports keywors DISTINCT and ALL after SELECT` or something similar.

**Comment by JosefDohnal** [comment]
Yes I can, It's OK?

**Comment by milo** [comment]
It's not, there must be only one commit among this pull request, now it's eight.

**Comment by dg** [comment]
Merged b3b6e9bb73702119a596b4f408bad4356fc53cb1

**Pull request closed without merging**